pmd-java/src/main/java/net/sourceforge/pmd/lang/java/JavaLanguageModule.java

```diff
@@ -62,7 +58,12 @@ public LanguageProcessor createProcessor(LanguagePropertyBundle bundle) {
         return new JavaLanguageProcessor((JavaLanguageProperties) bundle);
     }
 
-    public static Language getInstance() {
-        return LanguageRegistry.PMD.getLanguageByFullName(NAME);
+    @Override
+    public Tokenizer createCpdTokenizer(LanguagePropertyBundle bundle) {
+        return new JavaTokenizer((JavaLanguageProperties) bundle);
+    }
+
+    public static JavaLanguageModule getInstance() {
+        return (JavaLanguageModule) LanguageRegistry.PMD.getLanguageByFullName(NAME);
```

Comment:
Sometimes we return the instance from the LanguageRegistry, sometimes we return a static field from the language module (e.g. in HtmlLanguageModule)... is there any difference?

Reply:
I've decided to make the getInstance() method consistent across all language modules: We should always return the instance from the language registry - that way, we most likely get the very same instance, regardless whether the language is queried via language registry or via getInstance(). Otherwise we would have at least 2 different instances of the languages. So this approach to query always the registry should safe one byte or two of memory.

Reply:
Actually this works for all languages except for Apex: Since Visualforce depends on Apex, we need a Apex Instance being available before the registry is there.. So I kept the duplicated apex instance.